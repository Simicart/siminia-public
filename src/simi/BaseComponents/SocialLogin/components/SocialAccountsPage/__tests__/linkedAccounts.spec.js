import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import LinkedAccounts from '../linkedAccounts';

jest.mock('../LinkedAccount', () => 'LinkedAccount');
jest.mock('react-intl', () => ({
  FormattedMessage: props => (
    <div componentName="Formatted Message Component" {...props} />
  ),
  useIntl: jest.fn().mockReturnValue({
    formatMessage: jest
      .fn()
      .mockImplementation(options => options.defaultMessage)
  })
}));

test('render empty', () => {
  const props = {
    accounts: null,
    handleUnlink: () => {},
    isDisabled: false
  };

  const tree = createTestInstance(<LinkedAccounts {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

test('render list', () => {
  const props = {
    accounts: [
      {
        label: 'Facebook',
        type: 'facebook',
        url:
          'https://ce241-pwa--atakarski.ap74.corp.amdev.by/amsociallogin/social/login/?type=facebook'
      },
      {
        label: 'Instagram',
        type: 'instagram',
        url:
          'https://ce241-pwa--atakarski.ap74.corp.amdev.by/amsociallogin/social/login/?type=instagram'
      }
    ],
    handleUnlink: () => {},
    isDisabled: false
  };

  const tree = createTestInstance(<LinkedAccounts {...props} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
