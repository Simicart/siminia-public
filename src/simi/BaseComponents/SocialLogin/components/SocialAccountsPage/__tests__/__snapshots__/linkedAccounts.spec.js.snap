// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render empty 1`] = `null`;

exports[`render list 1`] = `
<div>
  <div>
    <div
      componentName="Formatted Message Component"
      defaultMessage="Linked accounts"
      id="amSocialLogin.linkedAccounts"
    />
  </div>
  <ul>
    <LinkedAccount
      handleUnlink={[Function]}
      isDisabled={false}
      label="Facebook"
      type="facebook"
      url="https://ce241-pwa--atakarski.ap74.corp.amdev.by/amsociallogin/social/login/?type=facebook"
    />
    <LinkedAccount
      handleUnlink={[Function]}
      isDisabled={false}
      label="Instagram"
      type="instagram"
      url="https://ce241-pwa--atakarski.ap74.corp.amdev.by/amsociallogin/social/login/?type=instagram"
    />
  </ul>
</div>
`;
